### PR TITLE
Charger site area  cannot change limits when charging station not assigne to site area

### PR DESCRIPTION
**Pull request by OliveGerste**
I added checks to see, if a siteArea is assigned to the charging station.

But I found another inconsistency:
When you edit the charging station and it is not assigned to a site area, the site area prop appears mandatory and you can basically not edit the station. 
image

### CODE REVIEW

**Comment by LucasBrazi06** [comment]
@OliveGerste  Have you also tested the case where the charger may not be assigned to a site area when the organization component is disabled?

File: src/app/pages/charging-stations/charging-station-limitation/charging-plans/charging-plans.component.html [hunk 1]
```diff
@@ -5,24 +5,24 @@
         {{"chargers.smart_charging.charging_station_inactive" | translate }}
       </div>
     </div>
-    <div class="row" *ngIf="!chargingStation.inactive && isSmartChargingComponentActive && chargingStation.siteArea.smartCharging">
+    <div class="row" *ngIf="!chargingStation.inactive && isSmartChargingComponentActive && chargingStation.siteArea && chargingStation.siteArea.smartCharging">
```
Comment (LucasBrazi06) [inline src/app/pages/charging-stations/charging-station-limitation/charging-plans/charging-plans.component.html:8]: For (chargingStation.siteArea && chargingStation.siteArea.smartCharging), you can also use the shortcut with the question mark like this: (chargingStation.siteArea?.smartCharging)

Comment (LucasBrazi06) [reply src/app/pages/charging-stations/charging-station-limitation/charging-plans/charging-plans.component.html:8]: Update the code accordingly, thanks!

Comment (OliveGerste) [reply src/app/pages/charging-stations/charging-station-limitation/charging-plans/charging-plans.component.html:8]: Ah good to know. Thanks :)

Comment (LucasBrazi06) [reply src/app/pages/charging-stations/charging-station-limitation/charging-plans/charging-plans.component.html:8]: You welcome :-)

**Comment by OliveGerste** [comment]
> @OliveGerste Have you also tested the case where the charger may not be assigned to a site area when the organization component is disabled?

Yes I also checked, if it works, when the organization component is disabled.

**Comment by LucasBrazi06** [comment]
> > @OliveGerste Have you also tested the case where the charger may not be assigned to a site area when the organization component is disabled?
> 
> Yes I also checked, if it works, when the organization component is disabled.

Perfect 👍  !
Thanks Tobias !

**Pull request merged**